### PR TITLE
Cleanup-Round 1

### PR DESCRIPTION
**Pull request by azabbasi**
In this PR there are many types of code cleanups
* Removing dead code 
  * Unused methods
  * Unused classes
  * Unused fields and properties
* Formatting and refactoring

I have been cautious not to touch any sort of public-facing code. They should be removed on separate PR so we can take note of what we have changed in the surface API

There will be multiple rounds of these sorts of cleanups

### CODE REVIEW

**Review by timtay-microsoft — APPROVED** [review]

File: iothub/device/src/Common/ReadOnlyDictionary45.cs [hunk 1]
```diff
@@ -594,50 +581,51 @@ object ICollection.SyncRoot
...
         {
-            private readonly ICollection<TValue> m_collection;
+            private readonly ICollection<TValue> _collection;
 
             [NonSerialized]
-            private object m_syncRoot;
+            private object _syncRoot;
+
+            /// <summary>
+            ///
```
Comment (drwill-ms) [inline iothub/device/src/Common/ReadOnlyDictionary45.cs:590]: Fill in

File: iothub/service/src/Common/TimeoutHelper.cs [hunk 2]
```diff
@@ -1,19 +1,18 @@
 // Copyright (c) Microsoft. All rights reserved.
```
Comment (drwill-ms) [inline iothub/service/src/Common/TimeoutHelper.cs:1]: Can we get rid of TimeoutHelper?

File: iothub/service/src/Common/Timestamp.cs [hunk 3]
```diff
@@ -1,137 +1,110 @@
...
-    using System.Diagnostics;
-
     // This is meant to replace DateTime when it is primarily used to determine elapsed time. DateTime is vulnerable to clock jump when
     // system wall clock is reset. Stopwatch can be used in similar scenario but it is not optimized for memory foot-print.
     //
     // This class is immune to clock jump with the following two exceptions:
     //  - When multi-processor machine has a bug in BIOS/HAL that returns inconsistent clock tick for different processor.
     //  - When the machine does not support high frequency CPU tick.
-    struct Timestamp : IComparable<Timestamp>, IEquatable<Timestamp>
+    internal struct Timestamp : IComparable<Timestamp>, IEquatable<Timestamp>
```
Comment (drwill-ms) [inline iothub/service/src/Common/Timestamp.cs:15]: Do we still need this file/class?

File: iothub/service/src/Configurations/Configuration.cs [hunk 4]
```diff
@@ -56,11 +56,8 @@ internal Configuration()
         /// The key-value pairs used to describe the configuration.
         /// </summary>
         [JsonProperty(PropertyName = "labels", NullValueHandling = NullValueHandling.Ignore)]
-#pragma warning disable CA2227 // Collection properties should be read only
         public IDictionary<string, string> Labels { get; set; } = new Dictionary<string, string>();
```
Comment (drwill-ms) [inline iothub/service/src/Configurations/Configuration.cs:59]: Can we make the property readonly?

File: iothub/service/src/JobProperties.cs [hunk 5]
```diff
@@ -84,8 +82,6 @@ public JobProperties()
         [JsonProperty(PropertyName = "outputBlobContainerUri", NullValueHandling = NullValueHandling.Ignore)]
         public string OutputBlobContainerUri { get; set; }
```
Comment (drwill-ms) [inline iothub/service/src/JobProperties.cs:83]: Can we remove this `Uri` suffix so it doesn't imply a type?

File: iothub/service/src/JobProperties.cs [hunk 6]
```diff
@@ -143,8 +139,6 @@ public JobProperties()
         [JsonProperty(PropertyName = "configurationsBlobName", NullValueHandling = NullValueHandling.Ignore)]
         private string ConfigurationsBlobName { get; set; } // waiting for bug fix before publicizing
```
Comment (drwill-ms) [inline iothub/service/src/JobProperties.cs:140]: Oops, this comment can be removed

File: provisioning/service/src/Auth/ServiceAuthenticationWithSharedAccessPolicyToken.cs [hunk 7]
```diff
@@ -20,20 +20,20 @@ internal sealed class ServiceAuthenticationWithSharedAccessPolicyToken : IAuthen
...
+            SetPolicyName(policyName);
+            SetToken(token);
         }
 
         public string PolicyName
         {
-            get { return this.policyName; }
-            set { this.SetPolicyName(value); }
+            get { return policyName; }
+            set { SetPolicyName(value); }
```
Comment (drwill-ms) [inline provisioning/service/src/Auth/ServiceAuthenticationWithSharedAccessPolicyToken.cs:30]:
```suggestions
get => polcyName;
set => SetPolicyName(value);
```

File: provisioning/service/src/Config/AttestationMechanism.cs [hunk 8]
```diff
@@ -179,7 +179,7 @@ private TpmAttestation Tpm
 
                 /* SRS_ATTESTATION_MECHANISM_21_003: [If the provided attestation is instance of TpmAttestation, the
```
Comment (drwill-ms) [inline provisioning/service/src/Config/AttestationMechanism.cs:180]: Shall we remove these comments, like we do in Java?

File: provisioning/service/src/Auth/ServiceAuthenticationWithSharedAccessPolicyToken.cs (hunk 7 above)
Comment (azabbasi) [reply provisioning/service/src/Auth/ServiceAuthenticationWithSharedAccessPolicyToken.cs:30]: Already taken care of in the second round of cleanup.

File: iothub/service/src/Common/TimeoutHelper.cs (hunk 2 above)
Comment (azabbasi) [reply iothub/service/src/Common/TimeoutHelper.cs:1]: We will, once we get rid of all the API flavors that use it.

File: iothub/service/src/Common/Timestamp.cs (hunk 3 above)
Comment (azabbasi) [reply iothub/service/src/Common/Timestamp.cs:15]: hehe ! we don't ! will remove it.

**Pull request merged**